pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -252,6 +252,10 @@ private void receiveMessageFromConsumer(ConsumerImpl<T> consumer, boolean batchR
             messagesFuture = consumer.receiveAsync().thenApply(Collections::singletonList);
         }
         messagesFuture.thenAcceptAsync(messages -> {
+            if (consumer.isDuringSeek()) {
+                receiveMessageFromConsumer(consumer, batchReceive);
```

Comment:
When `incomingMessages` is not empty, there are two possibilities:
1. loop until there is no message in the `incomingMessages`
2. If `ConsumerImpl.seek` fails, the messages have popped from `incomingMessages` cannot be consumed until `redeliver` executes

Reply:
I also have a question about this part.

It looks like not only from the MultiTopicConsumer. If it is from the user side, using a consumer to receive the messages and another thread try to seek the subscription to another position, they will also receive the duplicated messages right?

Reply:
> 2\. If `ConsumerImpl.seek` fails, the messages have popped from `incomingMessages` cannot be consumed until `redeliver` executes

Good catch! We need to consider this.

> It looks like not only from the MultiTopicConsumer.

Yes. 

>  If it is from the user side, using a consumer to receive the messages and another thread try to seek the subscription to another position, they will also receive the duplicated messages right?

For `MultiTopicConsumer`, this is right, because the `MultiTopicConsumer` has a loop to pulling the messsage.

Reply:
Ok, I guess the fix can only fix the case that the MultiTopicConsumer poll messages from the internal consumer, but it can't fix the issue that the user facing. After the message polled from the internal consumer to the queue of the MultiTopicConsumer, the user will still have chance to get duplicated messages during the seek operation?

Reply:
> Ok, I guess the fix can only fix the case that the MultiTopicConsumer poll messages from the internal consumer.

Right.

> the user will still have chance to get duplicated messages during the seek operation?

I think we need to figure out the details of this seek. 

1. When to clean `incomingMessages`, after or before seek?
2. During the seek, can the client continue to consume the `incomingMessages`?